hal/TARGET_FLASH_CMSIS_ALGO/flash_data.h

```diff
@@ -18,35 +18,44 @@
 
 #include <stdint.h>
 
-// Target flash algorithm structure
+/** Target flash algorithm structure
+ */
 typedef struct {
-    const uint32_t init;
-    const uint32_t uninit;
-    const uint32_t erase_sector;
-    const uint32_t program_page;
-    const uint32_t static_base;
-    uint32_t *algo_blob;
+    const uint32_t init; /**< Init function address */
+    const uint32_t uninit; /**< Uninit function address */
+    const uint32_t erase_sector; /**< Erase sector function address */
+    const uint32_t program_page; /**< Program page function address */
+    const uint32_t static_base; /**< Static base address */
+    uint32_t *algo_blob; /**< Pointer to flash algo binary blob */
 } flash_algo_t;
 
+/** Sector information structure
+ */
 typedef struct {
-    const uint32_t start;
-    const uint32_t size;
+    const uint32_t start; /**< Sector start address */
+    const uint32_t size; /**< Sector size */
 } sector_info_t;
 
+/** Flash configuration structure
+ */
 typedef struct {
-    const uint32_t page_size;
-    const uint32_t flash_start;
-    const uint32_t flash_size;
-    const sector_info_t *sectors;
-    const uint32_t sector_info_count;
+    const uint32_t page_size; /**< The minimum program page size */
+    const uint32_t flash_start; /**< Program page size */
+    const uint32_t flash_size; /**<  Flash size */
+    const sector_info_t *sectors; /**<  List of sectors */
+    const uint32_t sector_info_count; /**< Number of sectors */
```

Comment:
The meanings of `sector` and `page` are so ambiguous unless you define the relationship among `sector` , `page` and `flash_size` by an equation.

Reply:
I added more info there